api/server/app.cc

```diff
@@ -440,34 +442,58 @@ static const char *SrcCgroup() {
     return NULL;
 }
 
+void catFiles(std::string file1, std::string file2, std::string logMsg){        
+	std::ifstream ifs(file1);
+	std::ofstream ofs(file2);
+    if((ifs.is_open()) && (ofs.is_open())){
+        ofs << ifs.rdbuf();
+        ifs.close();
+        ofs.close();
+    } else {
+        printf("%s",(char*)logMsg.c_str());
+    }
+}
+
 static int InitCgroup(int multiplier) {
     const char *cgroupPath = SrcCgroup();
-
+    int check = 0;
+    
     if (!cgroupPath)
         return -1;
-    std::string create_dir("mkdir " + std::string(cgroupPath) + "/butterfly");
-
-    BASH(create_dir) {
-        LOG_WARNING_("can't create butterfly cgroup, fail cmd '%s'",
-                     create_dir.c_str());
+    std::string directory = std::string(cgroupPath) + "/butterfly" ; 
+    check = mkdir((char*)directory.c_str(),S_IWGRP );
+    if (check != 0){
+	    printf("can't create directory already exists\n");
     }
-    BASH("echo $(( `cat " + std::string(cgroupPath) + "/cpu.shares` * " +
-         std::to_string(multiplier) + " )) > " +
-         cgroupPath + "/butterfly/cpu.shares") {
-        LOG_WARNING_("can't set cgroup priority");
+    
+    std::string fileToRead = std::string(cgroupPath) + "/cpu.shares";
+    std::string fileToWrite = std::string(cgroupPath) + "/butterfly/cpu.shares";
+    std::ifstream in(fileToRead);
+    std::ofstream out(fileToWrite);
+    if ((in.is_open()) && (out.is_open())){
```

Comment:
I prefer to handle the error branch at the begin of the function and exit if it happen